src/agent/thread_helper.cpp

```diff
@@ -172,9 +172,9 @@ void ThreadHelper::Attach(const std::string &         aNetworkName,
     {
         uint64_t mask = UINT8_MAX;
 
-        for (int i = sizeof(uint64_t) - 1; i >= 0; i--)
+        for (size_t i = 0; i < sizeof(uint64_t); i++)
         {
-            extPanId.m8[i] = static_cast<uint8_t>((aExtPanId >> (i * CHAR_BIT)) & mask);
+            extPanId.m8[i] = static_cast<uint8_t>((aExtPanId >> ((sizeof(uint64_t) - i - 1) * CHAR_BIT)) & mask);
```

Comment:
```suggestion
            extPanId.m8[i] = static_cast<uint8_t>((aExtPanId >> ((sizeof(uint64_t) - i - 1) * 8)) & mask);
```
`8` is better than `CHAR_BIT` because we are explicitly using `uint8_t`, not `unsigned char`.